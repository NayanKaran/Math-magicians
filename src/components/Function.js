import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

export default class Function extends React.PureComponent {
    render = () => {
      const buttons = [];
      const { functions: { values, clickEventHandler, id } } = this.props;
      values.forEach((value) => {
        buttons.push(
          <Button
            key={value}
            buttonProp={{
              onClick: clickEventHandler,
              value,
            }}
          />,
        );
      });

      return (
        <div id={id}>
          {buttons}
        </div>
      );
    };
}

Function.propTypes = {
  functions: PropTypes.shape({
    values: PropTypes.arrayOf(PropTypes.string),
    clickEventHandler: PropTypes.func,
    id: PropTypes.string,
  }),
};

Function.defaultProps = {
  functions: null,
};
