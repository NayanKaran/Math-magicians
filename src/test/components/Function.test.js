import renderer from 'react-test-renderer';
import Function from '../../components/Function';

describe('Testing Function component', () => {
  const functions = {
    values: ['AC', '+/-', '%'],
    clickEventHandler: jest.fn(),
    id: 'unary-functions',
  };
  it('Testing with snapshot', () => {
    const tree = renderer.create(<Function functions={functions} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
