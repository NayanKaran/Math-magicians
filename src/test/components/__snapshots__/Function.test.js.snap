// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Function component Testing with snapshot 1`] = `
<div
  id="unary-functions"
>
  <button
    onClick={[MockFunction]}
    type="button"
    value="AC"
  >
    AC
  </button>
  <button
    onClick={[MockFunction]}
    type="button"
    value="+/-"
  >
    +/-
  </button>
  <button
    onClick={[MockFunction]}
    type="button"
    value="%"
  >
    %
  </button>
</div>
`;
